fastparse/shared/src/main/scala/fastparse/parsers/Intrinsics.scala

```diff
@@ -16,8 +17,8 @@ object Intrinsics {
     private[this] val uberSet = CharBitSet(chars)
     def parseRec(cfg: ParseCtx, index: Int) = {
       val input = cfg.input
-      if (index >= input.length) fail(cfg.failure, index)
-      else if (uberSet(input(index))) success(cfg.success, (), index + 1, Nil, false)
+      if (Try(input.charAt(index)).isFailure) fail(cfg.failure, index)
+      else if (uberSet(input.charAt(index))) success(cfg.success, (), index + 1, Nil, false)
```

Comment:
@lihaoyi not completely happy with having an intermediate exception being generated here, but .length is too eager. any better ideas?